Firestore/Source/Core/FSTView.mm

```diff
@@ -328,23 +338,23 @@ - (FSTViewDocumentChanges *)computeChangesWithDocuments:(const MaybeDocumentMap
 
     if (changeApplied) {
       if (newDoc) {
-        newDocumentSet = [newDocumentSet documentSetByAddingDocument:newDoc];
+        newDocumentSet = newDocumentSet.insert(newDoc);
         if (newDoc.hasLocalMutations) {
           newMutatedKeys = newMutatedKeys.insert(key);
         } else {
           newMutatedKeys = newMutatedKeys.erase(key);
         }
       } else {
-        newDocumentSet = [newDocumentSet documentSetByRemovingKey:key];
+        newDocumentSet = newDocumentSet.erase(key);
         newMutatedKeys = newMutatedKeys.erase(key);
       }
     }
   }
 
   if (self.query.limit) {
-    for (long i = newDocumentSet.count - self.query.limit; i > 0; --i) {
-      FSTDocument *oldDoc = [newDocumentSet lastDocument];
-      newDocumentSet = [newDocumentSet documentSetByRemovingKey:oldDoc.key];
+    for (long i = newDocumentSet.size() - self.query.limit; i > 0; --i) {
```

Comment:
s/long/size_t/ ?
(This will eventually throw a compiler warning/error once ported to linux/gcc due to unsigned->signed conversion.)

Hm. Actually, scratch that. You're subtracting query.limit; this could legitimately be negative. Or it could still overflow.

"Fixing" this likely requires adding some noise. Feel free to ignore/defer.